mesmer/mesmer_m/power_transformer.py

```diff
@@ -290,3 +291,233 @@ def inverse_transform(self, transformed_monthly_T, yearly_T):
+
+    # the computation of lambda is influenced by NaNs so we need to
+    # get rid of them
+    local_monthly_residuals = local_monthly_residuals[
+        ~np.isnan(local_monthly_residuals)
+    ]
+    local_yearly_T = local_yearly_T[~np.isnan(local_yearly_T)]
+
+    def _neg_log_likelihood(coeffs):
+        """Return the negative log likelihood of the observed local monthly residual temperatures
+        as a function of lambda."""
+        lambdas = lambda_function(coeffs, local_yearly_T)
+
+        # version with own power transform
+        transformed_local_monthly_resids = _yeo_johnson_transform_np(
+            local_monthly_residuals, lambdas
+        )
+
+        n_samples = local_monthly_residuals.shape[0]
+        loglikelihood = -n_samples / 2 * np.log(transformed_local_monthly_resids.var())
+        loglikelihood += (
+            (lambdas - 1)
+            * np.sign(local_monthly_residuals)
+            * np.log1p(np.abs(local_monthly_residuals))
+        ).sum()
+
+        return -loglikelihood
+
+    bounds = np.array([[0, np.inf], [-0.1, 0.1]])
+    first_guess = np.array([1, 0])
+
+    xi_0, xi_1 = minimize(
+        _neg_log_likelihood,
+        x0=first_guess,
+        bounds=bounds,
+        method="Nelder-Mead",
+    ).x
+
+    return xi_0, xi_1
+
+
+def _get_lambdas_from_covariates_xr(coeffs, yearly_T):
+
+    lambdas = 2 / (1 + coeffs.xi_0 * np.exp(yearly_T * coeffs.xi_1))
+
+    return lambdas.rename("lambdas")
+
+
+def fit_power_transformer_xr(monthly_residuals, yearly_T):
+    """Estimate the optimal parameter lambda for each gridcell, given an xarray dataset
+    of temperature residuals.
+    The optimal lambda parameter for minimizing skewness is estimated on
+    each gridcell independently using maximum likelihood.
+
+    Parameters
+    ----------
+    monthly_residuals : xr.DataArray of shape (n_years*12, n_gridcells)
+        Monthly residuals after removing harmonic model fits, used to fit for the optimal transformation parameters (lambdas).
+
+    yearly_T :  xr.DataArray of shape (n_years, n_gridcells)
+        yearly temperature values used as predictors for the lambdas.
+
+    Returns
+    -------
+    :obj:`xr.DataSet`
+        Dataset containing the estimate coefficients xi_0 and xi_1 needed to estimate lambda with dimensions (months, n_gridcells) and the lambdas themselves with dimensions (months, n_gridcells, n_years).
+
+    """
+    monthly_residuals_grouped = monthly_residuals.groupby("time.month")
+
+    coeffs = []
+    for month in range(1, 13):
+        xi_0, xi_1 = xr.apply_ufunc(
+            _yeo_johnson_optimize_lambda,
+            monthly_residuals_grouped[month],
+            yearly_T,
+            input_core_dims=[["time"], ["time"]],
+            output_core_dims=[[], []],
+            output_dtypes=[float, float],
+            vectorize=True,
+            join="outer",
+        )
+        data = {"xi_0": xi_0, "xi_1": xi_1}
```

Comment:
```suggestion
        data_vars = {"xi_0": xi_0, "xi_1": xi_1}
```

(or create the `Dataset` directly)